app/services/dangling_account_remover.rb

```diff
@@ -0,0 +1,13 @@
+# frozen_string_literal: true
+
+# Service for removing dangling accounts
+class DanglingAccountRemover
+  class << self
+    def run!
+      Account.where.not(
+        account_number: Host.with_policies_or_test_results
+                            .select(:account).distinct
+      ).delete_all
+    end
```

Comment:
I may have misled you to point you to the dangling accounts metric. No compliance hosts is only one thing we should check. We definitely also want to check if the account has any policies, even if the policies don't have hosts. There is also no foreign key relationship between profiles and accounts, so this could leave some profiles pointing to nonexistent accounts. Finally, there may be some users associated to this account - those should probably just be removed along with the account if all the above conditions hold.